chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -1070,3 +1070,27 @@ func (b *BitcoindNotifier) CancelMempoolSpendEvent(
 
 	b.memNotifier.UnsubscribeEvent(sub)
 }
+
+// LookupInputMempoolSpend takes an outpoint and queries the mempool to find
+// its spending tx. Returns the tx if found, and boolean to indicate whether it
+// exists or not.
+//
+// NOTE: part of the MempoolWatcher interface.
+func (b *BitcoindNotifier) LookupInputMempoolSpend(
+	op wire.OutPoint) (*wire.MsgTx, bool) {
```

Comment:
Consider having this return an Option

Reply:
yeah i like it, updated in the following commit